generic/CTensor.c

```diff
@@ -0,0 +1,261 @@
+  else if( (src = luaT_toudata(L, 2, "torch.CudaDoubleTensor")) )
+    THTensor_(copyCudaDouble)(cutorch_getstate(L), tensor, src);
+  else
+    luaL_typerror(L, 2, "torch.*Tensor");
+
+  lua_settop(L, 1);
+  return 1;
+}
+
+static int TH_CONCAT_3(cutorch_,Real,Tensor_copyAsyncCuda)(lua_State *L)
+{
+  THTensor *storage = luaT_checkudata(L, 1, "torch.FloatTensor");
+  void *src;
+  if( (src = luaT_toudata(L, 2, "torch.CudaTensor")) )
+    THTensor_(copyAsyncCuda)(cutorch_getstate(L), storage, src);
+  else
+    luaL_typerror(L, 2, "torch.CudaTensor");
+
+  lua_settop(L, 1);
+  return 1;
+}
+
+
+
+#ifdef THC_REAL_IS_FLOAT
+static void THFloatTensor_computesz(THFloatTensor *self, long **sz_, long **st_)
+{
+  long *sz, *st, *szh;
+  int i;
+
+  sz = (long*)THAlloc(sizeof(long)*self->nDimension);
+  st = (long*)THAlloc(sizeof(long)*self->nDimension);
+  szh = (long*)THAlloc(sizeof(long)*self->nDimension);
+
+  for(i = self->nDimension-1; i >= 0; i--)
+  {
+    if(i == self->nDimension-1)
+      szh[i] = 1;
+    else
+      szh[i] = szh[i+1]*self->size[i+1];
+  }
+
+  memcpy(sz, szh, self->nDimension * sizeof(long));
+  memcpy(st, self->stride, self->nDimension * sizeof(long));
+  THFree(szh);
+
+  *sz_ = sz;
+  *st_ = st;
+}
+
+void THFloatTensor_kernel_copy(float *dst,
+                                         long *dst_sz, long *dst_st, int dst_dim,
+                                         float *src,
+                                         long *src_sz, long *src_st, int src_dim,
+                                         long n_elem)
+{
+  long k;
+
+  for(k = 0; k < n_elem; k++)
+  {
+    long src_idx = 0;
+    long src_rest = k;
+    long dst_idx = 0;
+    long dst_rest = k;
+    int dim;
+
+    for(dim = 0; dim < dst_dim; dim++)
+    {
+      dst_idx += (dst_rest/dst_sz[dim])*dst_st[dim];
+      dst_rest = dst_rest % dst_sz[dim];
+    }
+
+    for(dim = 0; dim < src_dim; dim++)
+    {
+      src_idx += (src_rest/src_sz[dim])*src_st[dim];
+      src_rest = src_rest % src_sz[dim];
+    }
+
+    dst[dst_idx] = src[src_idx];
+  }
+}
+
+static int cuda_FloatTensor_fakecopy(lua_State *L)
+{
+  THFloatTensor *self = luaT_checkudata(L, 1, "torch.FloatTensor");
+  THFloatTensor *src = luaT_checkudata(L, 2, "torch.FloatTensor");
+  long *d_self_sz, *d_self_st, *d_src_sz, *d_src_st;
+  long nElement = THFloatTensor_nElement(self);
+
+  THArgCheck(THFloatTensor_nElement(self) == THFloatTensor_nElement(src), 2, "sizes do not match");
+
+  THFloatTensor_computesz(self, &d_self_sz, &d_self_st);
+  THFloatTensor_computesz(src, &d_src_sz, &d_src_st);
+
+  THFloatTensor_kernel_copy(THFloatTensor_data(self),
+                            d_self_sz, d_self_st, self->nDimension,
+                            THFloatTensor_data(src),
+                            d_src_sz, d_src_st, src->nDimension,
+                            nElement);
+
+  THFree(d_self_sz);
+  THFree(d_self_st);
+  THFree(d_src_sz);
+  THFree(d_src_st);
+
+  lua_settop(L, 1);
+  return 1;
+}
```

Comment:
Do you know why this is here? I realize it was there before, but I wonder what it's for :)